mypy/stubtest.py

```diff
@@ -1526,7 +1526,52 @@ def is_read_only_property(runtime: object) -> bool:
     return isinstance(runtime, property) and runtime.fset is None
 
 
-def safe_inspect_signature(runtime: Any) -> inspect.Signature | None:
+def safe_inspect_signature(
+    runtime: Any, object_path: list[str] | None = None
+) -> inspect.Signature | None:
+    if (
+        hasattr(runtime, "__name__")
+        and runtime.__name__ == "__init__"
```

Comment:
Might be worth checking if something similar applies to `__new__`. I remember stubdefaulter had a similar problem with `__init__` on C classes that also affected `__new__`

Reply:
Here's the stubdefaulter issue: https://github.com/JelleZijlstra/stubdefaulter/issues/97

Reply:
This does work but will need finessing; I think typeshed policy is to not add the `__new__` when we already have `__init__`, and adding this for `__new__` creates a lot of errors where that's the case. Needs some logic to suppress those.